README.md

```diff
@@ -26,104 +26,142 @@ What is wrong with the classic format?
-disambiguate binary data like `author` or `previous` from ordinary
-byte strings.
-
-Message payload; an array with 5 elements in this specific order:
-
-1) `author` a binary [SSB-BFE] encoded feed id
-2) `sequence` a 32 bit integer starting at 1
-3) `previous` a binary [SSB-BFE] encoded message id of the previous message
-  on the feed. For the first message this must be zero bytes.
-4) `time` an integer representing the UNIX epoch timestamp of message
-  creation
-5) `content` an array of a dictionary encoding the data relevant to the
-  meta feed and a signature. The signature is the concatenation of the
-  string 'metafeeds' encoded as bytes and the bytes of the content
-  payload, signed using the private key of the sub feed. If content is
-  encrypted this will also be encrypted as a binary [SSB-BFE] encoded
-  box2 message
+## Definitions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC 2119.
+
+We shall use the notation `[a, b, c]` to denote a [bencode] *list*,
+`{a: 1, b: 2}` to denote a [bencode] *dictionary*, and the notation `a + b + c`
+to denote byte concatenation.
+
+## Specification
+
+A *Bendy Butt message* is encoded in [bencode] as a list of message `payload`
+and `signature`, which we shall denote `[payload, signature]`. SSB Binary Field
+Encodings ([SSB-BFE]) is used to encode all non-list non-dictionary non-integer
+values contained in the `payload` and `signature`. BFE also helps to
+disambiguate binary data like feed IDs and message IDs from generic data types
+such as strings.
+
+The message `payload` is a bencode list with 5 elements in this specific order,
+in other words, `[author, sequence, previous, timestamp, contentSection]`:
+
+1) `author`, a BFE-encoded feed ID
+2) `sequence`, a 32 bit integer greater than or equal to 1
+3) `previous`, a BFE-encoded message ID of the previous message
+  on the feed. For the first message this must be the BFE "nil" generic data
+  type.
+4) `timestamp`, a 32 bit integer representing the UNIX epoch timestamp of message
+  creation.
+5) `contentSection` is `[content, contentSignature]` such that:
+    1. `content`, any bencode dictionary representing the message contents, all
+    of which are internally encoded with BFE.
+    2. `contentSignature`, a BFE-encoded "signature". The signature is applied
+    on `prefix + content`, where prefix is the string "bendybutt" encoded as
+    UTF8 bytes (i.e. the `prefix`), using some cryptographic keypair (this
+    **MAY** be different from the `author` keypair). If `content` is encrypted,
+    this will also be encrypted as a BFE-encoded box2 message.
+
+The message `signature` is applied on `prefix + payload`, where `prefix` is the
+string "bendybutt" encoded as UTF8 bytes, same prefix as used in
+`contentSignature`, signed with the `author`'s cryptographic keypair.
```

Comment:
We only need the prefix for the contentSignature. This is for domain separation as the key signing that part might be signing messages on their feed as well and we want to make sure that someone can never take a signature from that and put it into a bendy butt message.

Reply:
Good to know!

Reply:
Resolved but I don't want this subthread to be collapsed.